server/src/main/java/org/opensearch/common/settings/FeatureFlagSettings.java

```diff
@@ -35,6 +35,7 @@ protected FeatureFlagSettings(
         new HashSet<>(
             Arrays.asList(
                 FeatureFlags.REPLICATION_TYPE_SETTING,
+                FeatureFlags.SEGMENT_REPLICATION_CLUSTER_REPLICATION_TYPE_SETTING,
```

Comment:
This could be renamed REPLICATION_TYPE_SEGMENT_REPLICATION_CLUSTER_SETTING for readability. That way we preserve the SEGMENT_REPLICATION_CLUSTER_SETTING in the name to show users it's related.

Reply:
Sure will update this in next commit.